src/main/java/nl/tudelft/contextproject/Main.java

```diff
@@ -91,10 +121,13 @@ public static void setInstance(Main main) {
 	public boolean setController(Controller c) {
 		if (c != controller) {
 			if (controller != null) {
-				stateManager.detach(controller);
+				getStateManager().detach(controller);
 			}
 			controller = c;
-			stateManager.attach(controller);
+			
+			if (controller != null) {

```

Comment:
Looks like I either forgot how to read or my git diff was broken, as I swear line 122 said c != null